main.cpp

```diff
@@ -1867,7 +1868,7 @@ struct picoboot_memory_access : public memory_access {
     }
 
     void read(uint32_t address, uint8_t *buffer, unsigned int size, __unused bool zero_fill) override {
-        if (flash == get_memory_type(address, model)) {
+        if (flash == get_memory_type(address, model) && settings.use_flash_cache) {
```

Comment:
Is it worth swapping these conditionals around, as I guess if `settings.use_flash_cache` returns false then there's no need to check `flash == get_memory_type(address, model)` ?